### PR TITLE
Feat: Download all videos together

### PR DESCRIPTION
**Pull request by dixidroid**
Added the option to download all videos at once on the video tab.

https://github.com/openedx/openedx-app-android/assets/5153976/55401f90-ba50-4de8-b327-2356c5668c17

### CODE REVIEW

File: core/src/main/java/org/openedx/core/presentation/settings/VideoQualityFragment.kt [hunk 1]
```diff
@@ -75,13 +80,19 @@ class VideoQualityFragment : Fragment() {
             OpenEdXTheme {
                 val windowSize = rememberWindowSize()
 
-                val videoQuality by viewModel.videoQuality.observeAsState(viewModel.currentVideoQuality)
+                val title = stringResource(
+                    id = if (viewModel.getQualityType() == VideoQualityType.Streaming)
+                        R.string.core_video_streaming_quality else
+                        R.string.core_video_download_quality
+                )
```
Comment (volodymyr-chekyrta) [inline core/src/main/java/org/openedx/core/presentation/settings/VideoQualityFragment.kt:87]: Difficult to read construction.
Not required, but `else` could be moved to a separate line.

**Comment by volodymyr-chekyrta** [comment]
Not sure about moving `ProfileNotifier` to the core module.
Could you please explain the purpose of doing this? Some event needs to be consumed outside of the profile?

File: course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt [hunk 2]
```diff
@@ -292,6 +295,77 @@ fun CourseSectionCard(
     }
 }
 
+@Composable
+fun OfflineQueueCard(
+    downloadModel: DownloadModel,
+    progressValue: Long,
+    progressSize: Long,
+    onDownloadClick: (DownloadModel) -> Unit
+) {
```
Comment (volodymyr-chekyrta) [inline course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt:304]: Could you please add a preview for this Card?

File: course/src/main/java/org/openedx/course/settings/download/DownloadQueueFragment.kt [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+}
+
+
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_NO, device = Devices.TABLET)
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_YES)
+@Composable
+private fun DownloadQueueScreenPreview() {
+    DownloadQueueScreen(
+        windowSize = WindowSize(WindowType.Compact, WindowType.Compact),
+        uiState = DownloadQueueUIState.Models(
```
Comment (volodymyr-chekyrta) [inline course/src/main/java/org/openedx/course/settings/download/DownloadQueueFragment.kt:222]: Should we put the DownloadQueueScreen inside the OpenEdXTheme?
```
OpenEdXTheme {
    DownloadQueueScreen(
        ...
    )
}
```

File: core/src/main/java/org/openedx/core/AppDataConstants.kt [hunk 4]
```diff
@@ -10,4 +10,6 @@ object AppDataConstants {
 
     const val VIDEO_FORMAT_M3U8 = ".m3u8"
     const val VIDEO_FORMAT_MP4 = ".mp4"
-}
\ No newline at end of file
+
+    const val DOWNLOADS_CONFIRMATION_SIZE = 1073741824L
```
Comment (volodymyr-chekyrta) [inline core/src/main/java/org/openedx/core/AppDataConstants.kt:14]: Could you please explain what this size means?

Comment (dixidroid) [reply core/src/main/java/org/openedx/core/AppDataConstants.kt]: this is equal to 1 GB, added comment for explanation

File: course/src/main/java/org/openedx/course/presentation/videos/CourseVideosFragment.kt
[the remark this answers is not in the document]
Comment (dixidroid) [reply course/src/main/java/org/openedx/course/presentation/videos/CourseVideosFragment.kt:1]: done

File: course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt (hunk 2 above)
Comment (dixidroid) [reply course/src/main/java/org/openedx/course/presentation/ui/CourseUI.kt:305]: done

File: core/src/main/java/org/openedx/core/presentation/settings/VideoQualityFragment.kt (hunk 1 above)
Comment (dixidroid) [reply core/src/main/java/org/openedx/core/presentation/settings/VideoQualityFragment.kt]: done

**Review by k1rill — APPROVED** [review]

**Pull request merged**